### PR TITLE
feat(alerts): Add timeline view to alert rule details

### PR DESCRIPTION
**Pull request by taylangocmen**
This adds a timeline of incidents to the alert rule details page.
Currently all incidents are shown in the timeline but they might be limited by the chart window as well in the future.
Also will need to include `seenBy` results in the incidents query.

[Jira - WOR-595](https://getsentry.atlassian.net/browse/WOR-595)
[Design](https://www.figma.com/file/9bqoJ1hNQZWudZUNjf72qe/Handover%3A-Metric-Alerts-Detail?node-id=27%3A10)

Multiple Incidents:
Screen Shot 2021-02-11 at 2 38 41 AM

Random Incident with weird activity:
Screen Shot 2021-02-11 at 2 36 42 AM

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    let Icon = IconCheckmark;
+    let color: string = theme.green300;
+
+    if (
+      activities &&
+      activities.find(
+        activity =>
+          activity.type === IncidentActivityType.DETECTED ||
+          (activity.type === IncidentActivityType.STATUS_CHANGE &&
+            activity.value === `${IncidentStatus.CRITICAL}`)
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx:206]: could do `activities?.find`

There's something too long about these to be inside of the if statement

File: src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+    try {
+      const activities = await fetchIncidentActivities(api, orgId, incident.id);
+      this.setState({activities, loading: false});
+    } catch (err) {
+      this.setState({loading: false, error: !!err});
+    }
+  }
+
+  renderActivity(activity: ActivityType, idx) {
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx:79]:
```suggestion
  renderActivity(activity: ActivityType, idx) {
  renderActivity(activity: ActivityType, idx: number) {
```

File: src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx (hunk 1 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx:206]: or maybe filter by ```activity.type === IncidentActivityType.STATUS_CHANGE &&
            activity.value === `${IncidentStatus.CRITICAL}```` and then find on that

Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx]: I shortened it up a bit

**Review by scttcper — APPROVED** [review]

**Pull request merged**